### PR TITLE
Add bindings to CompileOnDemandLayer

### PR DESCRIPTION
**Pull request by cocreature**
This is basically the equivalent of `IRCompileLayer` but it compiles modules lazily.

### CODE REVIEW

File: llvm-general/src/LLVM/General/Internal/OrcJIT/IRCompileLayer.hs [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE MultiParamTypeClasses #-}
+module LLVM.General.Internal.OrcJIT.IRCompileLayer where
+
+import           LLVM.General.Prelude
```
Comment (bscarlet) [inline llvm-general/src/LLVM/General/Internal/OrcJIT/IRCompileLayer.hs]: Please don't add whitespace after import like this.
In general, don't add whitespace just to line things up, and attempt to style stuff as in the prexisting code.

**Review by bscarlet — CHANGES_REQUESTED** [review]
Only the minor changes requested. Otherwise looks good.

File: llvm-general/src/LLVM/General/Internal/OrcJIT/IRCompileLayer.hs (hunk 1 above)
Comment (cocreature) [reply llvm-general/src/LLVM/General/Internal/OrcJIT/IRCompileLayer.hs]: Fixed, sorry for that. I need to figure out how to stop my emacs shortcut for sorting imports from aligning them.

**Comment by cocreature** [comment]
It’s been 3 weeks since I’ve fixed the issues you mentioned. Could you please merge this or tell me what needs to be changed?

**Comment by bscarlet** [comment]
Sorry. I've been recovering from a hardware failure at home & that's distracted from a lot of other projects.

**Pull request merged**